CMakeLists.txt

```diff
@@ -176,6 +188,25 @@ if (ENABLE_THREADS AND Threads_NOTFOUND)
   message(FATAL_ERROR "Cannot find pthreads (ENABLE_THREADS is ON).")
 endif ()
 
+# Build optional Intel HEXL library
+if(USE_INTEL_HEXL)
+  if(HEXL_DIR)
+    set(HEXL_DIR
+        ""
+        CACHE
+        STRING "Set location of HEXL cmake data for linking."
+        )
+    set(INIT_HEXL_DIR ${HEXL_DIR})
+  endif()
+  find_package(HEXL 1.2.0 HINTS ${HEXL_DIR} REQUIRED)
+  if(INIT_HEXL_DIR AND NOT INIT_HEXL_DIR STREQUAL HEXL_DIR)
+    message(WARNING "HEXL location provided '${INIT_HEXL_DIR}' \
+            does not equal chosen '${HEXL_DIR}'.")
+  endif()
+  include_directories(${HEXL_INCLUDE_DIRS})
```

Comment:
Better would be to include directories on a per-target basis, e.g. `target_include_directories(HElib ${HEXL_INCLUDE_DIRS})`. Same with `target_compile_definitions` vs `add_compile_definitions` below

Reply:
I agree and that was originally what I tried to use. However the helib target is not created until later on in the pipeline for some reason and I wanted to avoid any major changes to the CMakeLists.